aws-lambda/src/http.rs

```diff
@@ -0,0 +1,56 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use hyper::body::Bytes;
+use reqwest::Client;
+use tailcall::http::Response;
+use tailcall::HttpIO;
+
+#[derive(Clone)]
+pub struct LambdaHttp {
+    client: Client,
+}
+
+impl Default for LambdaHttp {
+    fn default() -> Self {
+        Self { client: Client::new() }
+    }
+}
+
+impl LambdaHttp {
+    pub fn init() -> Self {
+        Default::default()
+    }
+}
+
+#[async_trait::async_trait]
+impl HttpIO for LambdaHttp {
+    async fn execute(&self, request: reqwest::Request) -> Result<Response<Bytes>> {
+        let client = self.client.clone();
+        let method = request.method().clone();
+        let url = request.url().clone();
```

Comment:
do we need all this cloning? 
this in new allocations and cpu time to do cloning

Reply:
Fixed by pre-generating a string with `format!` for logging before `request` is consumed.